src/main/java/com/example/Storehouse/service/ResponseService.java

```diff
@@ -0,0 +1,170 @@
+                .idProduct(products.getIdProduct())
+                .name(products.getName())
+                .quantity(quantity)
+                .dateOfManufacture(products.getDateOfManufacture())
+                .daysToExpire(products.getDaysToExpire())
+                .build());
+    }
+
+    private void requestToUpdateProductQuantityInDatabase(String name) {
+        service.updateQuantityProduct(name);
+    }
+
+    private void requestToUpdateProductInDatabase(int quantity, String name) {
+        service.updateDataBase(name, quantity);
+    }
+
+    private void requestProcessedSuccessfully() {
+        print.info("Request was successfully processed\n");
+    }
+
+    private void addProductToList(String name, int quantity){
+            productList.add(buildProductForResponse(quantity, name).get());
+    }
+
+    private void clearListOfProducts(){
+        productList.clear();
+    }
+
+    private void requestToSaveRefundProductToDatabase(RefundProducts refundProduct){
+        service.saveRefundProduct(refundProduct);
+    }
+
+    private int requestToGetQuantityOfProductFromDatabase(String name){
+        return service.getQuantityOfProduct(name);
+    }
+
+
+     public  Optional<SingleProductResponse> getResponse(SingleProductRequest singleProductRequest) {
+        checkingTheProductInTheDatabase(singleProductRequest);
+        return Optional.ofNullable(SingleProductResponse.builder()
+                .nameOfProduct(buildProductForResponse(singleProductRequest.getProductQuantity(), singleProductRequest.getProductName()).get())
+                .requestUUID(singleProductRequest.getRequestUUID())
+                .build());
+    }
+    public  Optional<RefundResponse> getResponse(RefundRequest refundRequest){
+        saveProductInDatabase(refundRequest);
+        return Optional.ofNullable(RefundResponse.builder()
+                .requestUUID(refundRequest.getRequestUUID())
+                .message("Request was successfully processed")
+                .build());
+    }
+
+    public  Optional<CheckQuantityResponse> getResponse(CheckQuantityOfProductRequest request){
+        buildCheckQuantityProductResponse(request);
+        return Optional.ofNullable(CheckQuantityResponse.builder()
+                .requestUUID(request.getRequestUUID())
+                .nameOfProduct(request.getNameOfProduct())
+                .quantityOfProduct(requestToGetQuantityOfProductFromDatabase(request.getNameOfProduct()))
+                .build());
+    }
+
+
+
+    public  Optional<MultiProductsResponse> getResponse(MultiProductRequest multiProductRequest){
+        checkingTheProductInTheDatabase(multiProductRequest);
+        return Optional.ofNullable(MultiProductsResponse
+                .builder().products(productList)
+                .uuid(multiProductRequest.getRequestUUID())
+                .build());
+    }
+
+
+    private void checkingTheProductInTheDatabase(SingleProductRequest singleProductRequest) {
```

Comment:
This is very common mistake. you naming method checkingTheProductInTheDatabase, for me this method is Only checking product in database but in reality addition to that it also make modifications to objects in database, see the problem?

Reply:
done